pkgs/by-name/st/stirling-pdf/package.nix

```diff
@@ -0,0 +1,112 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  gradle_7,
+  perl,
+  makeWrapper,
+  jre,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "stirling-pdf";
+  version = "0.22.3";
+
+  src = fetchFromGitHub {
+    owner = "Stirling-Tools";
+    repo = "Stirling-PDF";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-8zXTapFAXw4+KLLirxBeEBmqNw6ILFHtbsaBSP3Ehyg=";
+  };
+
+  patches = [
+    # fix dependency resolution issues caused by spotless
+    ./remove-spotless.patch
+    # remove timestamp from the header of a generated .properties file
+    ./remove-props-file-timestamp.patch
+    # stripJavaArchivesHook doesn't patch signed JAR files, so instead
+    # use gradle's built-in method of zeroing out timestamps
+    ./fix-jar-timestamp.patch
+  ];
+
+  # fake build to pre-download deps into fixed-output derivation
+  deps = stdenv.mkDerivation {
+    name = "striling-pdf-${finalAttrs.version}-deps";
+    inherit (finalAttrs) src patches;
+
+    nativeBuildInputs = [
+      gradle_7
+      perl
+    ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      export GRADLE_USER_HOME=$(mktemp -d)
+      gradle --no-daemon --console=plain build
+
+      runHook postBuild
+    '';
+
+    # perl code mavenizes pathes (com.squareup.okio/okio/1.13.0/a9283170b7305c8d92d25aff02a6ab7e45d06cbe/okio-1.13.0.jar -> com/squareup/okio/okio/1.13.0/okio-1.13.0.jar)
+    installPhase = ''
+      runHook preInstall
+
+      find $GRADLE_USER_HOME/caches/modules-2 -type f -regex '.*\.\(jar\|pom\)' \
+        | perl -pe 's#(.*/([^/]+)/([^/]+)/([^/]+)/[0-9a-f]{30,40}/([^/\s]+))$# ($x = $2) =~ tr|\.|/|; "install -Dm444 $1 \$out/$x/$3/$4/$5" #e' \
+        | sh
+      # weird dependency resolution fix
+      ln -s $out/com/squareup/okio/okio-jvm/3.6.0/okio-jvm-3.6.0.jar $out/com/squareup/okio/okio/3.6.0/okio-3.6.0.jar
+
+      runHook postInstall
+    '';
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "recursive";
+    outputHash = "sha256-33bGMRfCQZesu031wXlr05wTAPcK92DfT5mKlMkMzSo=";
+  };
+
+  nativeBuildInputs = [
+    gradle_7
+    makeWrapper
+  ];
+
+  # use the generated maven repository
+  postPatch = ''
+    sed -i "1ipluginManagement { repositories { maven { url '${finalAttrs.deps}' } } }" settings.gradle
```

Comment:
I think it would be more readable to use a patch and substituteAll for this, like in https://git.hubrecht.ovh/hubrecht/nix-pkgs/src/branch/main/self/stirling-pdf/default.nix#L76

Reply:
Yeah, you're probably right. I'll see what I can do.

Reply:
I pushed the change